### PR TITLE
Comms set

### LINKED ISSUE #300: Duplicate foreign keys are not removed
Duplicate foreign keys cause comm/handler functions to also be duplicated in Go. Imagine it's a `Seq` vs `Set` issue somewhere

MRE:
```
Tinder: project {
  #language(go);
  #database(postgres);
  #provider(dockerCompose);
}

User: service {
  firstName: string;
}

Match: service {
  userOne: User;
  userTwo: User;
}
```

### PR DESCRIPTION
**Pull request by jaylees14**
All I wanted to do today was add `list` support into `temple test`, but I keep hitting problems 😢 

This fixes #300, where we treated foreign keys as a `seq`, rather than a set, causing duplication of code. Have also updated tests to include foreign keys, now we support them.

### CODE REVIEW

**Review by Lewiky — APPROVED** [review]
LGTM

File: src/test/scala/temple/builder/DatabaseBuilderTestData.scala [hunk 1]
```diff
@@ -5,7 +5,7 @@ import temple.generate.database.ast.ColumnConstraint.{Check, NonNull, Unique}
 import temple.generate.database.ast.ComparisonOperator.{GreaterEqual, LessEqual}
 import temple.generate.database.ast.Condition.PreparedComparison
 import temple.generate.database.ast.Statement._
-import temple.generate.database.ast._
+import temple.generate.database.ast.{ColumnDef, _}
```
Comment (c-harding) [inline src/test/scala/temple/builder/DatabaseBuilderTestData.scala]: huh?

**Review by c-harding — COMMENTED** [review]
looks good, nice and simple

**Review by c-harding — APPROVED** [review]

File: src/test/scala/temple/builder/DatabaseBuilderTestData.scala (hunk 1 above)
Comment (jaylees14) [reply src/test/scala/temple/builder/DatabaseBuilderTestData.scala]: I'm not sure either, shall fix

**Pull request merged**